utils/gitlab_api.py

```diff
@@ -293,22 +297,42 @@ def get_access_level(access):
         elif access == "guest":
             return gitlab.GUEST_ACCESS
 
+    def get_group_id_and_projects(self, group_name):
+        groups = self.gl.groups.list()
+        group = [g for g in groups if g.path == group_name]
```

Comment:
Do you ever expect more than one group here? If not, you should break the loop when you find your group.

Reply:
i do not. in a single gitlab instance there can only be one group with a given name.